src/storage/ducklake_metadata_manager.cpp

```diff
@@ -2097,32 +2097,97 @@ ORDER BY snapshot_id
 	return snapshots;
 }
 
-vector<DuckLakeFileScheduledForCleanup> DuckLakeMetadataManager::GetFilesScheduledForCleanup(const string &filter) {
-	auto res = transaction.Query(R"(
+vector<DuckLakeFileForCleanup> DuckLakeMetadataManager::GetOldFilesForCleanup(const string &filter) {
+	auto query = R"(
 SELECT data_file_id, path, path_is_relative, schedule_start
 FROM {METADATA_CATALOG}.ducklake_files_scheduled_for_deletion
-)" + filter);
+)" + filter;
+	auto res = transaction.Query(query);
 	if (res->HasError()) {
 		res->GetErrorObject().Throw("Failed to get files scheduled for deletion from DuckLake: ");
 	}
 	auto context = transaction.context.lock();
-	vector<DuckLakeFileScheduledForCleanup> result;
+	vector<DuckLakeFileForCleanup> result;
 	for (auto &row : *res) {
-		DuckLakeFileScheduledForCleanup info;
+		DuckLakeFileForCleanup info;
 		info.id = DataFileIndex(row.GetValue<idx_t>(0));
 		DuckLakePath path;
 		path.path = row.GetValue<string>(1);
 		path.path_is_relative = row.GetValue<bool>(2);
 		info.path = FromRelativePath(path);
 		info.time = GetTimestampTZFromRow(*context, row, 3);
-
 		result.push_back(std::move(info));
 	}
 	return result;
 }
+vector<DuckLakeFileForCleanup> DuckLakeMetadataManager::GetOrphanFilesForCleanup(const string &filter,
+                                                                                 const string &data_path) {
+	auto query = R"(SELECT filename
+FROM read_blob('{DATA_PATH}**')
+WHERE filename NOT IN (
+SELECT
+    CASE
+        WHEN NOT f.path_is_relative THEN f.path
+        ELSE
+            CASE
+                WHEN NOT t.path_is_relative THEN t.path || f.path
+                ELSE '{DATA_PATH}' || s.path || t.path || f.path
+            END
+    END AS full_path
+FROM ducklake_data_file f
+JOIN ducklake_table t ON f.table_id = t.table_id
+JOIN ducklake_schema s ON t.schema_id = s.schema_id
+UNION ALL
+SELECT
+    CASE
+        WHEN NOT f.path_is_relative THEN f.path
+        ELSE
+            CASE
+                WHEN NOT t.path_is_relative THEN t.path || f.path
+                ELSE '{DATA_PATH}' || s.path || t.path || f.path
+            END
+    END AS full_path
+FROM ducklake_delete_file f
```

Comment:
Maybe we can run the union of `ducklake_delete_file` and `ducklake_data_file` before the join so we don't have to duplicate these joins and this path handling?